src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,110 @@
+ * into the chatbot upon startup (or creating a new dir/file if it doesn't exist), and updating its
+ * contents after the user is done with the chatbot.
+ */
+public class Storage {
+    private final String path;
+    private final File savedTasks;
+
+    /**
+     * Constructor for Storage, which initialises the directory and file to save tasks in.
+     * @param path Filepath for the saved tasks file.
+     * @throws EkudIOException Exception involving the creation of the saved tasks file.
+     */
+    public Storage(String path) throws EkudIOException {
+        this.path = path;
+        File file = new File(path);
+        try {
+            if (!file.getParentFile().exists()) { // Create directory if it doesn't exist
+                file.getParentFile().mkdirs();
+            }
+            if (!file.exists()) { // Create file if it doesn't exist
+                System.out.println("Creating task file...");
+                file.createNewFile();
+                System.out.println("Task file created successfully");
+            }
+            this.savedTasks = file;
+        } catch (IOException e) {
+            throw new EkudIOException("Error with creating task file: " + e);
+        }
+    }
+
+    /**
+     * Loads saved tasks from the file into the chatbot's TaskList.
+     * @param taskList The chatbot's TaskList to load tasks into.
+     * @throws EkudIOException Exception involving improper loading of saved tasks into TaskList.
+     */
+    public void loadData(TaskList taskList) throws EkudIOException {
+        Parser parser = new Parser(); // For parsing dateTime
+        System.out.println("Loading up saved tasks...");
+        try {
+            Scanner scanner = new Scanner(this.savedTasks);
+            int curTask = 0;
+            while (scanner.hasNext()) {
+                // Saved tasks format eg:
+                // T | 0 | task1
+                // D | 1 | task2 | 1st Sep
+                // E | 0 | task 3 | 1st Sep 2pm | 3rd Sep 2pm
+                String[] taskDetails = scanner.nextLine().split(" \\| ");
+                String taskType = taskDetails[0];
+                boolean isDone = taskDetails[1].equals("X");
+                if (taskType.equals("T")) {
+                    taskList.addTask(new ToDo(taskDetails[2]));
+                } else if (taskType.equals("D")) {
+                    LocalDateTime dateTime = parser.parseSavedDateTime(taskDetails[3]);
+                    taskList.addTask(new Deadline(taskDetails[2], dateTime));
+                } else if (taskType.equals("E")) {
+                    LocalDateTime fromDateTime = parser.parseSavedDateTime(taskDetails[3]);
+                    LocalDateTime toDateTime = parser.parseSavedDateTime(taskDetails[4]);
+                    taskList.addTask(new Event(taskDetails[2], fromDateTime, toDateTime));
+                }
+                if (isDone) taskList.markDoneOnStart(curTask);
```

Comment:
The provided guide suggests writing it this way:
```suggestion
                if (isDone) {
                    taskList.markDoneOnStart(curTask);
                }
```